test/tandem/testTandemSimulator.js

```diff
@@ -472,6 +507,65 @@ describe('tandemSimulator.js', function() {
       expect(lastEvent.annotations[0].code).to.equal('tandem/basal/fabricated-from-new-day');
 
     });
+
+    it('a new-day event during a temp basal', function() {
+      var tempBasalStart = {
+            type: 'temp-basal',
+            subType: 'start',
+            percent: 0.5,
+            duration: 1800000
+          };
+      var tempBasalStop = {
+            type: 'temp-basal',
+            subType: 'stop',
+            time_left: 0
+          };
+      var temp = builder.makeTempBasal()
+        .with_time('2014-09-25T23:50:00.000Z')
+        .with_deviceTime('2014-09-25T23:50:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0)
+        .with_rate(0.5);
+      var newDay = builder.makeScheduledBasal()
+        .with_time('2014-09-26T00:00:00.000Z')
+        .with_deviceTime('2014-09-26T00:00:00')
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0)
+        .with_rate(1);
+      newDay.set('type', 'new-day');
+
+      var basal2 = builder.makeScheduledBasal()
```

Comment:
Maybe just add a comment here about how it's disjoint but you're just using it to force all the ones you're interested in to be simulated.